src/evaluation/term_methods.h

```diff
@@ -38,6 +37,13 @@
 
 namespace evaluation {
 
+/* A context to precompute heavy operations so we don't have to do that
+ * multiple times for a single evaluation
+ * NOTE: Only values that are being reused between terms should be added here */
+struct TermContext {
+    const std::array<uint64_t, magic_enum::enum_count<Player>()> pawnAttacks;
```

Comment:
Is the inclusion of only pawn attacks an intuitive one, or did you try including some other heavy operations?

Reply:
Current we only do non-position based computations with the pawn attacks. The other computations are based on the pieces positions, so those we can't precompute. But good observation :+1: